### PR TITLE
fix(shapes): [iOS] Fix shapes rendering issues

### PR DESCRIPTION
**Pull request by dr1rrb**
GitHub Issue: _private see below_

## Bugfix / refactor
The way that shapes was rendered on iOS has been fully refactored to properly follow the measure/arrange phases of WinUI/Uno.

## What is the current behavior?
Some shapes are not rendered properly on iOS, and sometimes they are no visible at all (cf. issue linked below).

## What is the new behavior?
All shapes behaves like WinUI

## PR Checklist
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] ~~Contains **NO** breaking changes~~ _Some API have been aligned to match the WinUI contract_
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). _cf. link below_
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Shapes/Rectangle.iOSmacOS.cs [hunk 1]
```diff
@@ -1,188 +1,55 @@
...
-			//Set default stretch value
-			this.Stretch = Stretch.Fill;
-
-#if __IOS__
-			// Background color is black by default, if and only if overriding Draw(CGRect rect).
-			base.BackgroundColor = SolidColorBrushHelper.Transparent.Color;
-#else
-			base.WantsLayer = true;
-			base.Layer.BackgroundColor = SolidColorBrushHelper.Transparent.Color;
-#endif
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Shapes/Rectangle.iOSmacOS.cs:47]: Where did this go ?

File: src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+				default:
+				case Stretch.Fill:
+					// size is already valid ... nothing to do!
+					break;
+
+				case Stretch.Uniform when pathArea.Width < pathArea.Height:
+					pathArea.Height = pathArea.Width;
+					break;
+
+				case Stretch.Uniform: // when pathArea.Width > pathArea.Height:
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs:121]: Comment not needed ?

File: src/Uno.UI/UI/Xaml/Shapes/Rectangle.iOSmacOS.cs (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Shapes/Rectangle.iOSmacOS.cs:47]: In the [base class `Shape`](https://github.com/unoplatform/uno/pull/2911/files/#diff-1b16792b910be7fdd3f4ca3219e6d7c2R26) (ok I might have forgotten it and just pushed a commit to restore it ...)

File: src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs:126]: just to explicit the case ... not really needed indeed

File: src/Uno.UI/UI/Xaml/Shapes/Shape.cs [hunk 3]
```diff
@@ -25,99 +30,123 @@ private protected double ActualStrokeThickness
...
+		public static DependencyProperty StrokeProperty { get; } = DependencyProperty.Register(
+			"Stroke",
+			typeof(Brush),
+			typeof(Shape),
+			new FrameworkPropertyMetadata(
+				defaultValue: null,
+#if LEGACY_SHAPE_MEASURE
+				propertyChangedCallback: (s, e) => ((Shape)s).OnStrokeUpdated((Brush)e.NewValue)
+#else
+				options: FrameworkPropertyMetadataOptions.AffectsArrange
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Shapes/Shape.cs:78]: # 👍

File: src/Uno.UI/UI/LayoutHelper.cs [hunk 4]
```diff
@@ -237,6 +237,20 @@ internal static double NumberOrDefault(this double value, double defaultValue)
 				: value;
 		}
 
+		[Pure]
+		internal static double FiniteOrDefault(this double value, double defaultValue)
+		{
+#if XAMARIN
+			return IsFinite(value)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/LayoutHelper.cs:244]: Why ?

**Comment by dr1rrb** [comment]
This as been moved to a feature branch https://github.com/unoplatform/uno/pull/2997

**Pull request closed without merging**